curator-framework/src/test/java/org/apache/curator/framework/imps/TestReconfiguration.java

```diff
@@ -558,6 +606,7 @@ private CuratorFramework newClient(String connectionString) {
     private CuratorFramework newClient(String connectionString, boolean withEnsembleProvider)
     {
         final AtomicReference<String> connectString = new AtomicReference<>(connectionString);
+        ensembleLatch = new CountDownLatch(1);
```

Comment:
I suggest to add a `newClient(String connectionString, boolean withEnsembleProvider, CountDownLatch)` overload to avoid using field - this is somehow global variable that is brittle.

Reply:
Added `newClient(String connectionString, CountDownLatch ensembleLatch)`. The `withEnsembleProvider` parameter will be `true` if `ensembleLatch != null` in this overload.